tests/Stormpath/Tests/Resource/ApplicationTest.php

```diff
@@ -387,6 +393,72 @@ public function testSendPasswordResetEmailWithAccountStore()
         $groupA->delete();
     }
 
+    public function testSendVerificationEmail()
+    {
+        $application = self::$application;
+
+        $directory = \Stormpath\Resource\Directory::instantiate(array('name' => 'dir' . md5(time())));
+        self::createResource(\Stormpath\Resource\Directory::PATH, $directory);
+
+        \Stormpath\Resource\AccountStoreMapping::create(
+            array('accountStore' => $directory, 'application' => $application)
+        );
+
+        // set directory policy to enable verification email workflow
+        $policy = $directory->getProperty('accountCreationPolicy');
+        $policy->verificationEmailStatus = 'ENABLED';
+        $policyResource = new Resource(self::$client->dataStore, $policy);
+        $result = self::$client->create($policyResource->href, $policyResource);
+        $this->assertEquals('ENABLED', $result->getProperty('verificationEmailStatus'));
+
+        $username = 'acc' . md5(time());
+        $emailAddress = $username . '@unknown123.kot';
+        $account = Account::instantiate(array(
+            'givenName' => 'Account Name',
+            'middleName' => 'Middle Name',
+            'surname' => 'Surname',
+            'username' => $username,
+            'email' => $emailAddress,
+            'password' => 'superP4ss'));
+
+        $result = $directory->createAccount($account);
+        $this->assertEquals($username, $result->username);
+        $this->assertEquals($emailAddress, $result->email);
+
+        try
+        {
+            $request = new VerificationEmailRequest($username);
+            $application->sendVerificationEmail($request);
+        }
+        catch(ResourceError $re)
+        {
+            $this->fail("Send verification email failed: ".$re->getErrorCode()." ".$re->getDeveloperMessage());
+        }
+
+        try
+        {
+            $request = new VerificationEmailRequest($emailAddress);
+            $application->sendVerificationEmail($request);
+        }
+        catch(ResourceError $re)
+        {
+            $this->fail("Send verification email failed: ".$re->getErrorCode()." ".$re->getDeveloperMessage());
+        }
+
+        try
+        {
+            $request = new VerificationEmailRequest($emailAddress, array('accountStore' => $directory));
+            $application->sendVerificationEmail($request);
+        }
+        catch(ResourceError $re)
+        {
+            $this->fail("Send verification email failed: ".$re->getErrorCode()." ".$re->getDeveloperMessage());
+        }
+
+        $account->delete();
+        $directory->delete();
```

Comment:
If the account is in the directory, you don't need to delete the account first. All resources that belong to the directory will be deleted when you delete the directory.